engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/util/service/query/ServiceQuery.scala

```diff
@@ -54,7 +52,7 @@ class ServiceQuery(modelData: ModelData) {
       val collector = QueryServiceInvocationCollector(Some(TestRunId(UUID.randomUUID().toString)), serviceName)
 
       val variablesPreparer = GlobalVariablesPreparer(definitions.expressionConfig)
-      val validationContext = variablesPreparer.emptyValidationContext(metaData)
+      val validationContext = variablesPreparer.validationContextWithLocalVariables(metaData, ctx.variables.mapValues(Typed.fromInstance))
```

Comment:
Maybe it would be better to pass ValidationContext explicitly (or just have option to do it).  Typed.fromInstance can be inprecise...

Reply:
:+1: 
ok, but mabye we should pass `localVariables: Map[String, (Any, TypingResult)]` instead of `Context` and `ValidationContext` ? 
WDYT?

Reply:
ok for me :)